src/components/Admin/custom/PostPreview.tsx

```diff
@@ -0,0 +1,47 @@
+import {Stack} from '@mui/material'
+import Grid from '@mui/material/Unstable_Grid2'
+import {FC, useState} from 'react'
+import {FormDataConsumer, ThemeProvider} from 'react-admin'
+
+import {Post} from '@/components/Posts/Post'
+import {PostDetail} from '@/components/Posts/PostDetail'
+import {theme} from '@/theme'
+
+export const PostPreview: FC = () => {
+  const [isDetailOpen, openDetail] = useState<boolean>()
```

Comment:
dalsi nit: pisal by som `useState(false)`, aby to nezacinalo na undefined, a typ si to inferuje. doesn't matter really